test/utils/cluster.go

```diff
@@ -0,0 +1,113 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strconv"
+
+	"k8s.io/klog/v2"
+)
+
+func EnsureCluster(location, clusterName string, numOfNodes int) error {
+	params := []string{
+		"container",
+		"clusters",
+		"describe",
+		clusterName,
+		"--zone", location,
+	}
+	out, err := exec.Command("gcloud", params...).CombinedOutput()
+	if err != nil {
+		klog.Infof("Cluster %s does not exist, creating instead.", clusterName)
+		return createCluster(location, clusterName, numOfNodes)
+	}
+
+	pattern := "currentNodeCount: "
+	startIndex := bytes.Index(out, []byte(pattern)) + len(pattern)       // The index immediately after the pattern.
+	endIndex := startIndex + bytes.Index(out[startIndex:], []byte("\n")) // The index after the pattern and before new line.
+	if startIndex == -1 || endIndex == -1 {
+		klog.Infof("Cannot find current node count. Delete and recreate cluster.")
+		return deleteAndCreateCluster(location, clusterName, numOfNodes)
+	}
+
+	gotNumOfNodes, err := strconv.Atoi(string(out[startIndex:endIndex]))
+	if err != nil || gotNumOfNodes != numOfNodes {
+		klog.Infof("Got cluster with %d nodes, expect %d. Delete and recreate cluster %s in %s.", gotNumOfNodes, numOfNodes, clusterName, location)
+		return deleteAndCreateCluster(location, clusterName, numOfNodes)
+	}
+	klog.Infof("Use existing cluster %s in zone %s with %d nodes", clusterName, location, numOfNodes)
+	return nil
+}
+
+func createCluster(location, clusterName string, numOfNodes int) error {
+	klog.Infof("Creating cluster %s in %s, numOfNodes=%d", clusterName, location, numOfNodes)
+	params := []string{
+		"container",
+		"clusters",
+		"create",
+		clusterName,
+		"--zone", location,
+		"--num-nodes", strconv.Itoa(numOfNodes),
+	}
+	if out, err := exec.Command("gcloud", params...).CombinedOutput(); err != nil {
+		return fmt.Errorf("failed creating cluster: %s, err: %v", out, err)
+	}
+	return nil
+}
+
+func DeleteCluster(location, clusterName string) error {
+	klog.Infof("Deleting cluster %s in %s", clusterName, location)
+	params := []string{
+		"container",
+		"clusters",
+		"delete",
+		clusterName,
+		"--zone",
+		location,
+		"--quiet",
+	}
+	if out, err := exec.Command("gcloud", params...).CombinedOutput(); err != nil {
+		return fmt.Errorf("failed deleting cluster: %s, err: %v", out, err)
+	}
+	return nil
+}
+
+func deleteAndCreateCluster(location, clusterName string, numOfNodes int) error {
+	if err := DeleteCluster(location, clusterName); err != nil {
+		return fmt.Errorf("failed delete and create cluster: %s, err: %v", clusterName, err)
+	}
+	if err := createCluster(location, clusterName, numOfNodes); err != nil {
+		return fmt.Errorf("failed delete and create cluster: %s, err: %v", clusterName, err)
+	}
+	return nil
+}
+
+func GetCredential(location, clusterName string) error {
```

Comment:
We should note that GetCredentials will overwrite the user's cluster credentials that are saved in their .config/kube/config director.

Reply:
Added in comments.